src/utils/validation.ts

```diff
@@ -18,6 +22,22 @@ export function formatValidationError(
   return error;
 }
 
+export function replaceErrorMessagePlaceholders(errors: {
```

Comment:
Again, no opinion, but just something to consider as an alternative. Would it be more straightforward to leverage the replace function as is used here?: https://github.com/alphagov/di-authentication-frontend/blob/ed8c21e5646ace4c45e9d6f78a791e2db715b574/src/components/prove-identity-callback/index.njk#L2

This would work simply by passing the max length variable into the template from the controller and letting the .njk file deal with swapping in the right number...?

Reply:
Hi @DTaylorGDS, I got some advice from the Design System Team and have just pushed a commit that addresses this issue. What I've done is

- Use Nunjucks filters when _instantiating_ the `govukCharacterCount` component (rather than doing the filter upstream)
- Creating a new `replaceErrorMessagePlaceholders` function within `validation.ts` that prepares an error list for use within the error summary (because the error summary expects a list it is not possible to use Nunjucks filters here)
- Introducing a `PlaceholderReplacement` interface
- Extracting the existing code that de-duped error messages into a function to make the code a bit easier to understand
- Introduce `validation.test.ts` and new unit tests for these functions

Because these are quite significant changes I've retested every support form (for both missing field and length exceeded errors).